filterprotobuffer_test.go

```diff
@@ -0,0 +1,97 @@
+package dapper_test
+
+import (
+	"github.com/dogmatiq/dapper"
+	"github.com/dogmatiq/dapper/internal/fixtures"
+	"google.golang.org/protobuf/reflect/protoreflect"
+	"regexp"
+	"strings"
+	"testing"
+	"time"
+)
+
+// matchProtoFormat works around non-deterministic behaviour introduced by the protobuf package.
+// They want to enforce nobody relies on the output being stable. It's not totally unreasonable but this library
+// is all about the output, so we really want to know when it changes.
+// See https://github.com/golang/protobuf/issues/1121
+func matchProtoFormat(actual, expected string) bool {
+	pattern := strings.ReplaceAll(regexp.QuoteMeta(expected), ": ", ":  ?")
+	ok, err := regexp.MatchString(pattern, actual)
+
+	if err != nil {
+		panic(err)
+	}
+	return ok
+}
+
+func TestFilterProtobufferFormats(t *testing.T) {
+	protoStub := &fixtures.Protostub{
+		FirstField:   "hello",
+		EnumField:    fixtures.Protoenum_FOO,
+		InnerMessage: &fixtures.ProtostubInner{InnerFirstField: "foo"},
+	}
+
+	protoStub.ProtoReflect().SetUnknown(protoreflect.RawFields("\x12\x07testing"))
+	_ = protoStub.String() // Trigger population of internal state to make sure it does not render
+
+	actual := dapper.Format(protoStub)
+	expected := strings.Join([]string{
+		"*fixtures.Protostub{",
+		`    first_field: "hello"`,
+		"    enum_field: FOO",
+		"    inner_message: {",
+		`        inner_first_field: "foo"`,
+		"    }",
+		`    2: "testing"`,
+		"}",
```

Comment:
```suggestion
		`*fixtures.Protostub{`,
		`    first_field: "hello"`,
		`    enum_field: FOO`,
		`    inner_message: {`,
		`        inner_first_field: "foo"`,
		`    }`,
		`    2: "testing"`,
		`}`,
```

Suggested using raw strings on every line as you've done elsewhere, easier to move lines around, copy/paste, etc.